zebrad/src/commands/start.rs

```diff
@@ -311,10 +312,9 @@ impl StartCmd {
                     .map_err(|e| eyre!(e)),
 
                 progress_result = &mut progress_task_handle => {
+                    info!("chain progress task exited");
                     progress_result
                         .expect("unexpected panic in the chain progress task");
-                    info!("chain progress task exited");
-                    Ok(())
```

Comment:
The progress task never exits, so the only way we get here is via a panic in that task. Since this block resumes the panic, it never returns a value.

I documented this in edf4948.